hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -481,21 +509,65 @@ def clean_timesheets(self, timesheets):
         return timesheets
 
     def delete_empty_lines(self, delete_empty_rows=False):
-        for name in self.line_ids.mapped('value_y'):
+        for name in list(set(self.line_ids.mapped('value_y'))):
             row = self.line_ids.filtered(lambda l: l.value_y == name)
             if row:
-                ts_row = self.timesheet_ids.filtered(
-                    lambda x: x.project_id.id == row[0].project_id.id
-                    and x.task_id.id == row[0].task_id.id
-                )
-                if delete_empty_rows and self.add_line_project_id:
+                row_0 = fields.first(row)
+                is_add_line = self.add_line_project_id == row_0.project_id \
+                    and self.add_line_task_id == row_0.task_id
+                if delete_empty_rows and is_add_line:
                     check = any([l.unit_amount for l in row])
                 else:
                     check = not all([l.unit_amount for l in row])
                 if check:
+                    ts_row = self.timesheet_ids.filtered(
+                        lambda t: t.project_id.id == row_0.project_id.id
+                        and t.task_id.id == row_0.task_id.id
+                    )
                     ts_row.filtered(
                         lambda t: t.name == empty_name and not t.unit_amount
                     ).unlink()
+                    self._sheet_write(
+                        'timesheet_ids', self.timesheet_ids.exists())
+                    if not ts_row.exists() and delete_empty_rows:
```

Comment:
Why do you need to check here that `ts_row.exists()`? At this point `ts_row` should be always persistent IMO.

Reply:
steps:
- ts_row = [A, B, C] where A is the only one that has empty name and not unit amount
- A.unlink()
- ts_row is still [A, B, C] whereas ts_row.exists() is [B, C]

Reply:
Ok. Then I don't understand the next line.

In case `not ts_row.exists()`, you execute  `self._sheet_write('line_ids', self.line_ids.exists() - row)`. 

I'm trying to see what is the difference of behavior when this code is not present. It seems to me that it works the same, with or without this code.
In what case this code is needed?

Reply:
It was needed to update line_ids. Maybe now it's not need anymore because it's not stored. I will check. But the necessity of update line_ids was because line_ids is used at the beginning of the loop.